src/lib/es2022.sharedmemory.d.ts

```diff
@@ -2,6 +2,20 @@ interface Atomics {
     /**
      * A non-blocking, asynchronous version of wait which is usable on the main thread.
      * Waits asynchronously on a shared memory location and returns a Promise
+     * @param typedArray A shared Int32Array or BigInt64Array.
+     * @param index The position in the typedArray to wait on.
+     * @param value The expected value to test.
+     * @param [timeout] The expected value to test.
      */
-    waitAsync(typedArray: BigInt64Array | Int32Array, index: number, value: bigint, timeout?: number): { async: false, value: "ok" | "not-equal" | "timed-out" } | { async: true, value: Promise<"ok" | "not-equal" | "timed-out"> };
+    waitAsync(typedArray: Int32Array, index: number, value: number, timeout?: number): { async: false, value: "ok" | "not-equal" | "timed-out" } | { async: true, value: Promise<"ok" | "not-equal" | "timed-out"> };

```

Comment:
Also, in the latest proposal, the return value looks like

```ts
{ async: false, value: "not-equal" | "timed-out" } | { async: true, value: Promise<"ok" | "timed-out"> }
```

But I couldn't test this.